src/dstack/_internal/core/models/runs.py

```diff
@@ -301,6 +301,81 @@ def duration(self) -> timedelta:
+            termination_reason=termination_reason,
+            exit_status=exit_code,
+        )
+        return values
+
+    @staticmethod
+    def _get_status_message(
+        status: JobStatus,
+        termination_reason: Optional[JobTerminationReason],
+        exit_status: Optional[int],
+    ) -> str:
+        if status == JobStatus.DONE:
+            return "exited (0)"
+        elif status == JobStatus.FAILED:
+            if termination_reason == JobTerminationReason.CONTAINER_EXITED_WITH_ERROR:
+                return f"exited ({exit_status})"
+            elif termination_reason == JobTerminationReason.FAILED_TO_START_DUE_TO_NO_CAPACITY:
+                return "no offers"
+            elif termination_reason == JobTerminationReason.INTERRUPTED_BY_NO_CAPACITY:
+                return "interrupted"
+            else:
+                return "error"
+        elif status == JobStatus.TERMINATED:
+            if termination_reason == JobTerminationReason.TERMINATED_BY_USER:
+                return "stopped"
+            elif termination_reason == JobTerminationReason.ABORTED_BY_USER:
+                return "aborted"
+        return status.value
+
+    @root_validator
+    def _error(cls, values) -> Dict:
+        try:
+            termination_reason = values["termination_reason"]
+        except KeyError:
+            return values
+        values["error"] = JobSubmission._get_error(termination_reason=termination_reason)
+        return values
+
+    @staticmethod
+    def _get_error(termination_reason: Optional[JobTerminationReason]) -> Optional[str]:
+        if termination_reason == JobTerminationReason.INSTANCE_UNREACHABLE:
+            return "instance unreachable"
+        elif termination_reason == JobTerminationReason.WAITING_INSTANCE_LIMIT_EXCEEDED:
+            return "waiting instance limit exceeded"
+        elif termination_reason == JobTerminationReason.VOLUME_ERROR:
+            return "waiting runner limit exceeded"
+        elif termination_reason == JobTerminationReason.GATEWAY_ERROR:
+            return "gateway error"
+        elif termination_reason == JobTerminationReason.SCALED_DOWN:
+            return "scaled down"
+        elif termination_reason == JobTerminationReason.INACTIVITY_DURATION_EXCEEDED:
+            return "inactivity duration exceeded"
+        elif termination_reason == JobTerminationReason.TERMINATED_DUE_TO_UTILIZATION_POLICY:
+            return "utilization policy"
+        elif termination_reason == JobTerminationReason.PORTS_BINDING_FAILED:
+            return "ports binding failed"
+        elif termination_reason == JobTerminationReason.CREATING_CONTAINER_ERROR:
+            return "runner error"
+        elif termination_reason == JobTerminationReason.EXECUTOR_ERROR:
+            return "executor error"
+        elif termination_reason == JobTerminationReason.MAX_DURATION_EXCEEDED:
+            return "max duration exceeded"
+        else:
+            return None
```

Comment:
Maybe define a mapping instead of a big if? Also I think we'll forget to update this if when adding new termination reasons unless there is a test that checks all reasons are mapped.

Reply:
Fixed, also added tests